lib/network.go

```diff
@@ -1880,16 +1893,63 @@ type MsgDeSoHeader struct {
+	// For all later versions, this field will default to a value of zero.
+	//
 	// The nonce that is used by miners in order to produce valid blocks.
 	//
 	// Note: Before the upgrade from HeaderVersion0 to HeaderVersion1, miners would make
 	// use of ExtraData in the BlockRewardMetadata to get extra nonces. However, this is
 	// no longer needed since HeaderVersion1 upgraded the nonce to 64 bits from 32 bits.
 	Nonce uint64
 
-	// An extra nonce that can be used to provice *even more* entropy for miners, in the
+	// ExtraNonce is only used for Proof of Work blocks, with MsgDeSoHeader versions 0 and 1.
+	// For all later versions, this field will default to zero.
+	//
+	// An extra nonce that can be used to provide *even more* entropy for miners, in the
 	// event that ASICs become powerful enough to have birthday problems in the future.
 	ExtraNonce uint64
+
+	// ProposerPublicKey is only used for Proof of Stake blocks, starting with MsgDeSoHeader
+	// version 2. For all earlier versions, this field will default to nil.
+	//
+	// The ECDSA public key of the validator who proposed this block.
+	ProposerPublicKey *PublicKey
+
+	// ProposerVotingPublicKey is only used for Proof of Stake blocks, starting with
+	// MsgDeSoHeader version 2. For all earlier versions, this field will default to nil.
+	//
+	// The BLS public key of the validator who proposed this block.
+	ProposerVotingPublicKey *bls.PublicKey
+
+	// ProposedInView is only used for Proof of Stake blocks, starting with MsgDeSoHeader
+	// version 2. For all earlier versions, this field will default to nil.
+	//
+	// The view in which this block was proposed.
+	ProposedInView uint64
+
+	// ValidatorsVoteQC is only used for Proof of Stake blocks, starting with MsgDeSoHeader
+	// version 2. For all earlier versions, this field will default to nil.
+	//
+	// This is a QC containing votes from 2/3 of validators weighted by stake.
+	ValidatorsVoteQC *QuorumCertificate
+
+	// ValidatorsTimeoutAggregateQC is only used for Proof of Stake blocks, starting with
+	// MsgDeSoHeader version 2. For all earlier versions, this field will default to nil.
+	//
+	// In the event of a timeout, this field will contain the aggregate QC constructed from
+	// timeout messages from 2/3 of validators weighted by stake, and proves that they have
+	// timed out. This value is set to null in normal cases where a regular block vote has
+	// taken place.
+	ValidatorsTimeoutAggregateQC *TimeoutAggregateQuorumCertificate
+
+	// ProposerVotePartialSignature is only used for Proof of Stake blocks, starting with
+	// MsgDeSoHeader version 2. For all earlier versions, this field will default to nil.
+	//
+	// The block proposer's partial BLS signature of the (ProposedInView, BlockHash) pair
+	// for the block. This signature proves that a particular validator proposed the block,
+	// and also acts as the proposer's vote for the block.
+	ProposerVotePartialSignature *bls.Signature
```

Comment:
Side-note: Just wanted to say your comments overall are awesome, man. They're really going to help us and future engineers figure out what we were thinking. Great stuff.